### ISSUE TITLE
Support ListItem avatar and icon components

### ISSUE THREAD

**Issue by carlobambo** [react-native-training/react-native-elements#243]

**Comment by binoy14** [comment]
@carlobambo could you please elaborate more on what do you mean?

**Comment by carlobambo** [comment]
Currently it supports avatar by passing Image's source attribute like `{ uri: imageUrl }`. What it is missing is Custom avatar and Custom icon components


```
<ListItem
  avatar={<CustomAvatar {...avatarProps} />}
  {...listItemProps}
/>

```

or this one will do

```
<ListItem
  avatarComponent={CustomAvatar}
  {...listItemProps}
/>
```

**Comment by binoy14** [comment]
@carlobambo Yes, it is being worked on. We recently added the avatar component and we are working on adding the custom avatar component in the ListItem in the next release.

**Comment by carlobambo** [comment]
@binoy14 that's good. I'll wait for the release then. Thanks.

**Comment by GregoryPotdevin** [comment]
Would be awesome. I'm using react-native-elements but had to fork ListItem to allow for custom avatar/icon components.

**Comment by Monte9** [comment]
@binoy14 wanna take the lead on this? No rush, but hopefully we can get this in for our next release `v0.11.3`.

**Comment by ShenghuGong** [comment]
I am eager to this feature,Hope to see him as soon as possible,Thx

**Comment by binoy14** [comment]
@Monte9 I can try to get this working

**Comment by xavier-villelegier** [comment]
Hey guys, what about this issue ? Are you still working on it ?

**Comment by Monte9** [comment]
Thanks for the PR @xavier-villelegier! 💯 

This enhancement is added in PR #393 and has been merged. 

This will be available to use in `v0.12.3`
